lib/src/models.ts

```diff
@@ -58,6 +59,12 @@ export interface Header {
   type: Type;
 }
 
+export interface QueryParamComponent {
+  name: string;
+  type: Type;
+  required: boolean;
```

Comment:
Instead of having a `required` field here, I would check whether `type.kind === "optional"`.

Reply:
changed